modules/calib3d/src/solvepnp.cpp

```diff
@@ -311,9 +311,32 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
 
     opoints_inliers.resize(npoints1);
     ipoints_inliers.resize(npoints1);
-    result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+
+    try
+    {
+        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
                       distCoeffs, rvec, tvec, useExtrinsicGuess,
                       (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    }
+    catch(...){
```

Comment:
I think it is better to have `catch (const cv::Exception& e)`.

And log also `e.what()`. This way, if another type of exception is thrown from `solvePnP` it will be logged.

Reply:
Ok. I agree with you.